samples/window/CMakeLists.txt

```diff
@@ -1,22 +1,48 @@
 cmake_minimum_required(VERSION 3.1)
 
-# determine the build type
-if(NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build (Debug or Release)" FORCE)
-endif()
-
 # project name
 project(window)
 
 # define the executable
 set(EXECUTABLE_NAME "window")
-add_executable(${EXECUTABLE_NAME} main.cpp)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../../cmake/modules")
 
+include("${CMAKE_SOURCE_DIR}/../../cmake/Config.cmake")
+
+if(LUG_COMPILER_MSVC)
+    if(CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
+        string(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+    else()
+        list(APPEND CMAKE_CXX_FLAGS "/W4")
+    endif()
+elseif(LUG_COMPILER_GCC OR LUG_COMPILER_CLANG)
+    list(APPEND CMAKE_CXX_FLAGS "-Wall -Wextra -Werror -std=c++14")
+    string(REPLACE ";" " " "CMAKE_CXX_FLAGS" "${CMAKE_CXX_FLAGS}")
```

Comment:
I took the code from base at that time, we edited it after but indeed I didn't apply the change here